cloudinit/sources/helpers/azure.py

```diff
@@ -1055,4 +1055,213 @@ def dhcp_log_cb(out, err):
     )
 
 
+class BrokenAzureDataSource(Exception):
+    pass
+
+
+class NonAzureDataSource(Exception):
+    pass
+
+
+class OvfEnvXml:
+    OVF_NAMESPACE = "http://schemas.dmtf.org/ovf/environment/1"
+    WA_NAMESPACE = "http://schemas.microsoft.com/windowsazure"
+
+    ns = {
+        "ovf": OVF_NAMESPACE,
+        "wa": WA_NAMESPACE,
+    }
+
+    def __init__(
+        self,
+        *,
+        username: Optional[str] = None,
+        password: Optional[str] = None,
+        hostname: Optional[str] = None,
+        custom_data: Optional[bytes] = None,
+        disable_ssh_password_auth: Optional[bool] = None,
+        public_keys: Optional[list] = None,
+        preprovisioned_vm: bool = False,
+        preprovisioned_vm_type: Optional[str] = None,
+    ) -> None:
+        self.username = username
+        self.password = password
+        self.hostname = hostname
+        self.custom_data = custom_data
+        self.disable_ssh_password_auth = disable_ssh_password_auth
+        self.public_keys: list = public_keys or []
+        self.preprovisioned_vm = preprovisioned_vm
+        self.preprovisioned_vm_type = preprovisioned_vm_type
+
+    def __eq__(self, other) -> bool:
+        return self.__dict__ == other.__dict__
+
+    def __repr__(self) -> str:
+        props = [
+            "username=%s" % self.username,
+            "password=%s" % self.password,
```

Comment:
This is probably a bad idea in case the object gets logged at some point.  I'll probably drop the repr altogether.